### ISSUE TITLE
Racetrack mode uses wide turn with corner avoidance between rows on non-rectangular fields

### ISSUE THREAD

**Issue by raxtor** [Courseplay/courseplay#4091]
I am not sure if it was reported but crossing between rows not uses shortest way and is doing some "L" maneuvers. Its red line in image.
![fsScreen_2019_08_16_01_25_58](https://user-images.githubusercontent.com/831266/63136800-f7541c00-bfd3-11e9-9b5f-f3c00b94c35d.png)

**Comment by pvaiko** [comment]
BTW, have you tried this with  AI Turn enabled?

**Comment by raxtor** [comment]
I lightly tested AI turns already but not with this pattern because there is still problem with course generation #4176 which was closed as duplicate of this one but imho its different problem.

Can you tell me pls what diference is between AI turn and CP turn. I tested today Case QuadTrack with SPSL9 plow and got into troubles with both turn systems Similar probelm for Class Xerion. CP has problem during reverse maneuver and getting stuck.

I will test combines asap and let you know.

P.S.: Btw I tested also FS helper for plowing today and it has no trouble during reversing and turning. There are other problems but this one works OK, thats why I am asking for difference between AI and CP turn.

**Comment by raxtor** [comment]
![fsScreen_2019_11_02_22_54_34](https://user-images.githubusercontent.com/831266/68077421-d70e4f00-fdc3-11e9-8574-f83cce5525f2.png)
AI turn active...  I will create new issue for this if needed..

**Comment by raxtor** [comment]
Tested AI for plowing and still same result. 
![fsScreen_2019_11_03_05_44_48](https://user-images.githubusercontent.com/831266/68080525-87e51000-fdfd-11e9-829c-7d5433d7030f.png)

**Comment by pvaiko** [comment]
1. reversing is a completely different issue, open an other one if it is still a problem.
2. You original problem was with combines, that is why I recommended trying the AI turns. AI turns only work with non-towed implements, like a combine or a tractor with a hitch mounted tool.

**Comment by pvaiko** [comment]
Oh, one more thing: I keep asking folks to point me to a real life video showing a tractor with a big plow reversing during fieldwork.

**Comment by raxtor** [comment]
Roger, I am know using big plows a little more and it has relatively good results for irregular fields with 2 around headlands and 1 skipped row as shows a) variant of FW pattern. 
AI turns ON and and start using implements soon/stop implments late. So I think plowing is useable as it is. Anyway I assume that tractor is using same procedure **like combines** for move between rows in headland part or?
Its hard to find complex video where you can see plow pattern, try this one: 
Reversing big plows: 
https://youtu.be/TneRVibutaA?t=284
Typical pattern for plowing in RL:
https://www.youtube.com/watch?v=TALHD6vPEbk
https://youtu.be/J300vkPlinQ
Small plow turning maneuver:
https://youtu.be/gfU1lTYYQW4?t=198
But big trailed plows dont use reverse manuever during filed works. Non-trailed plows is possible use reverse maenuver

Here are some fieldwork patterns:
![fieldwork_patterns](https://user-images.githubusercontent.com/831266/68529555-28f13080-0300-11ea-80b9-3b85e9bc0e9c.png)
CP is able simulate a) pattern but for combines is most frequent c) d) f) becuse u have pipe most of time over harvested field. Also b) pattern is very nice!

Here are some headland turns:
![Illustration-of-the-geometrical-representation-of-a-field-with-four-common-turn-types](https://user-images.githubusercontent.com/831266/68529527-e7f91c00-02ff-11ea-931a-7e7b843164d3.png)
Imho only 1st and last is useable for big plows in RL. 2nd and 3rd are good for non-trailed plows

..and some ineteresting patterns yet:
![plow13 jpg](https://user-images.githubusercontent.com/831266/68529974-6d7ecb00-0304-11ea-97a0-01fff9061cf9.jpg)
![plow14 jpg](https://user-images.githubusercontent.com/831266/68529979-71aae880-0304-11ea-9e6d-4b12eb32e85c.jpg)

**Comment by pvaiko** [comment]
Thanks for the videos, this is what I was looking for. The drawings I had already (all of them:). I may implement b) at some point, c) we have already.

**Comment by raxtor** [comment]
yw, c) is possible? Can you post screenhoot of fieldwork generator settings for that pls? 

Anyway biggest problem is moving between rows if there is set some skipping rows. When combine is going to next row start point directly, usually is crashing to other vehicles which still working in row. Vehicle should follow headland path if he is moving between rows. No idea if there is needed fix for field generator or something other. I think spiral mode it ganerates correctly and vehicle is following headland path always.. I assume it will remove also "L" turn maneuver issue.

spiral correct:
![fsScreen_2019_11_09_17_16_38](https://user-images.githubusercontent.com/831266/68531650-d4a47b80-0314-11ea-9a00-806ac0fc7b8e.png)
circle wrong:
![fsScreen_2019_11_09_17_16_10](https://user-images.githubusercontent.com/831266/68531656-da9a5c80-0314-11ea-9f33-a5a8c8d94980.png)

CP "Big plows" has 1 issue yet. 
Check turn maneuver in video (47+ sec)...
https://youtu.be/TALHD6vPEbk?t=47
CP is making plow side change too early so in sharp turns there is already plow in opppsite side and ofc tractor is blocked by own plow. 
I tested how it is working with default FS helper and he is doing turn manuver in very similar to video. Plow is up in neutral/transport postion during turn maneuver. Can you take a look into that problem?

**Comment by raxtor** [comment]
and sry I somehow missed "AI turns only work with non-towed implements"... but it has no impact to my last post .

**Comment by raxtor** [comment]
@pvaiko just I found some interesting RL turn maneuvers vids if you are interested:
https://youtu.be/GBezj51kap4?t=205
https://youtu.be/wlaAoxa_iAM?t=151
https://youtu.be/4zTiEeWjYOQ?t=204
https://youtu.be/XvJyPdTBGmA?t=271

tight headlands:
https://youtu.be/bcbeJdbyEKA?t=109 
https://youtu.be/woLcjNghixk?t=16
https://youtu.be/ay0TLe4qQUk?t=31
